### PR TITLE
feat: deno 1.31.1 compat

### PR DESCRIPTION
**Pull request by zifeo**

### CODE REVIEW

**Comment by zifeo** [comment]
@littledivy please review

File: codegen.ts [hunk 1]
```diff
@@ -202,6 +195,11 @@ const { symbols } = Deno.dlopen({
   darwin: uri + "lib${name}.dylib",
   windows: uri + "${name}.dll",
   linux: uri + "lib${name}.so",
+  solaris: uri + "lib${name}.so",
```
Comment (lino-levan) [inline codegen.ts:198]: This seems unrelated to the PR. Could you revert this and split this off into another PR?

**Review by lino-levan — CHANGES_REQUESTED** [review]
Outside of the one nitpick, this looks good to me.

File: codegen.ts (hunk 1 above)
Comment (lino-levan) [reply codegen.ts:198]: Maybe there was a change to `/x/plug`. This seems to me to not be super future-proof. I think we should be using [name guessing](https://github.com/denosaurs/plug#example-using-automatic-binary-name-guessing) here.

**Comment by zifeo** [comment]
Closed in favour of #118

**Pull request closed without merging**